File: android/src/main/kotlin/com/screenmeet/sdk_live_flutter_plugin/handlers/LocalVideoStreamHandler.kt
package com.screenmeet.sdk_live_flutter_plugin.handlers

import com.screenmeet.sdk.ParticipantMediaState
import com.screenmeet.sdk_live_flutter_plugin.FlutterRTCVideoRenderer
import com.screenmeet.sdk_live_flutter_plugin.zipLocalVideo
import io.flutter.plugin.common.EventChannel

class LocalVideoStreamHandler: EventChannel.StreamHandler {

    private var sink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) { sink = events }
    override fun onCancel(arguments: Any?) { sink = null }

    fun sendLocalVideo(localMediaState: ParticipantMediaState, renderer: FlutterRTCVideoRenderer?) {
        val textureId = renderer?.textureId ?: -1L
        sink?.success(zipLocalVideo(localMediaState, textureId))
    }
}

File: ios/Classes/StreamHandlers/LocalVideoStreamHandler.swift
//
//  LocalVideoStreamHandler.swift
//  sdk_live_flutter_plugin
//
//  Created by Ross on 05.11.2021.
//

import Foundation
import Flutter
import ScreenMeetSDK

class LocalVideoStreamHandler: NSObject, FlutterStreamHandler {
    private let pm = SwiftChannelParamsManager()

    var sink: FlutterEventSink?
       
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }
       
    func sendLocalVideo(_ localMediaState: SMParticipantMediaState, _ renderer: FlutterRTCVideoRenderer?) {
        guard let sink = sink else { return }
        sink(pm.localVideo(localMediaState, renderer))
    }
       
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
}

